src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx

```diff
@@ -671,37 +671,30 @@ export class PrebuiltPredictPage extends React.Component<IPrebuiltPredictPagePro
 
             const extendPredictionItem = (key, field) => {
                 const result = {};
+                const updateFieldValueToResult = (item, itemName) => {
+                    if (item.valueObject) {
+                        Object.keys(item.valueObject).forEach(key => {
+                            result[`${itemName}: ${key}`] = item.valueObject[key];
+                        });
+                    }
+                    else {
+                        result[itemName] = item;
+                    }
+                }
+
                 if (field.valueArray) {
                     if (field.valueArray.length === 1) {
```

Comment:
suggestion: I would prefer simplify a bit more as the if-else are only different in the first itemName. We can easily achieve the same result with field.valueArray.forEach and checking the item index in the foreach.

Reply:
like this?
```
                    field.valueArray.forEach((item, index) => {
                        const itemName = field.valueArray.length === 1 ? key : `${key} ${index + 1}`;
                        updateFieldValueToResult(item, itemName);
                    });
```

Reply:
yes :)